rosetta/services/block.go

```
@@ -34,80 +34,79 @@ func (s *BlockAPIService) Block(
+			if !status.IsSynced() {
+				// When not fully synced, we cannot know if the requested TipSet
+				//is synced within the node
+				return nil, ErrUnableToGetUnsyncedBlock
+			}
+			reqCid, _ := cid.Parse(*request.BlockIdentifier.Hash)
+			requestedTipSetKey = filTypes.NewTipSetKey(reqCid)
 		}
+	} else {
+		return nil, ErrMalformedValue
 	}
 
-	var block *filTypes.BlockHeader
-	var err error
-
 	if useHeight {
-		tipset, err := s.node.ChainGetTipSetByHeight(ctx, abi.ChainEpoch(requestedHeight), filTypes.EmptyTSK)
-		if err != nil {
-			return nil, ErrUnableToGetTipset
-		}
-
-		block, err = s.node.ChainGetBlock(ctx, tipset.Cids()[0]) //TODO see issue https://github.com/Zondax/rosetta-filecoin/issues/1
-		if err != nil {
-			return nil, ErrUnableToGetBlk
-		}
-
+		tipSet, err = s.node.ChainGetTipSetByHeight(ctx, abi.ChainEpoch(requestedHeight), filTypes.EmptyTSK)
 	} else {
-		block, err = s.node.ChainGetBlock(ctx, requestedCid)
-		if err != nil {
-			return nil, ErrUnableToGetBlk
-		}
+		tipSet, err = s.node.ChainGetTipSet(ctx, requestedTipSetKey)
 	}
 
-	parentBlocks := block.Parents
-
-	if len(parentBlocks) == 0 {
-		return nil, ErrUnableToGetParentBlk
-	}
-
-	parentBlock, err := s.node.ChainGetBlock(ctx, parentBlocks[0])
 	if err != nil {
-		return nil, ErrUnableToGetBlk
+		return nil, ErrUnableToGetTipset
 	}
 
-	blockMsgs, err := s.node.ChainGetBlockMessages(ctx, block.Cid())
-	if err != nil {
-		return nil, ErrUnableToGetTxns
+	if tipSet.Parents().IsEmpty() {
+		return nil, ErrUnableToGetParentBlk
 	}
 
-	var txs []*types.Transaction
-
-	for _, msg := range blockMsgs.BlsMessages {
-		txs = append(txs, &types.Transaction{
-			TransactionIdentifier: &types.TransactionIdentifier{
-				Hash: strconv.FormatUint(msg.Nonce, 10),
-			},
-			Operations: nil, //TODO check this
-			Metadata:   nil, //TODO check this
-		})
+	var transactions []*types.Transaction
+	for _, block := range tipSet.Blocks() {
+		messages,_ := s.node.ChainGetBlockMessages(ctx, block.Cid())
```

any reason for ignoring the error here?

---

Missed it, added!